numba_cuda/numba/cuda/tests/cudadrv/test_cuda_driver.py

```diff
@@ -285,6 +285,58 @@ def b2d(bs):
         self.assertTrue(grid > 0)
         self.assertTrue(block > 0)
 
+    def test_cuda_set_shared_memory_carveout(self):
+        from numba import types
+        import numpy as np
+
+        sig = (types.float32[::1], types.float32[::1])
+
+        @cuda.jit(sig)
+        def add_one(r, x):
+            i = cuda.grid(1)
+            if i < len(r):
+                r[i] = x[i] + 1
+
+        kernel = add_one.overloads[sig]
+        cufunc = kernel._codelibrary.get_cufunc()
+
+        # valid carveout values
+        carveout_values = [0, 50, 100]
```

Comment:
Is it worth testing with the default as well?


```suggestion
        carveout_values = [-1, 0, 50, 100]
```